File: 1024-video-stitching/1024-video-stitching.java
class Solution {
    public int videoStitching(int[][] clips, int time) {
        int start = 0;
        int end = 0;
        int count = 0;

        while(start < time) {
            for(int[] c : clips){
                if(start >= c[0]){
                    end = Math.max(end, c[1]);
                }
            }

            if(start == end){
                return -1;
            }
            count ++;
            start = end;
        }

        return count;
        
    }
}

File: 1024-video-stitching/1024-video-stitching.py
class Solution:
    def videoStitching(self, clips: List[List[int]], time: int) -> int:
        clips.sort()
        start, end, count = 0, 0, 0
        i = 0
        n = len(clips)

        while start < time:
            while i < n and clips[i][0] <= start:
                end = max(end, clips[i][1])
                i += 1
            if start == end:
                return -1 

            count += 1
            start = end
            
        return count

        